### ISSUE TITLE
Double the training set with self-play from the current bot, relabelled at depth 10

### ISSUE THREAD

**Issue by stefanobaghino** [stefanobaghino/chessbot#51]
Every bucketed-NNUE ablation on #44 came out neutral or negative with the current 4.1M unique positions (data/all_unique.npz), and the last unbucketed retrain (#12) gained from depth-10 labels rather than from architecture. The next lever is more data from positions the current bot actually reaches.

Plan (long-running, resumable, cores 2-3 inside 09:00-21:00):

1. Self-play with the v0.1.45 engine (NNUE net6, 5k nodes per move, UHO book, 2 games per opening) via scripts/selfplay_gen.sh in the `quiet` cgroup: 4 batches of 10k games, about 3M positions after pgn_to_fens.py's sampling. Timed spars freeze the cgroup, so this can run alongside the queued #45 and #44 batches tomorrow and continue afterwards.
2. Relabel at Stockfish depth 10 with scripts/relabel_chunks.sh (resumable chunks, window-aware, exit 3 outside the window), queued behind the spars with scripts/queue.sh. Measured rate on #12: 50k positions per 34 min per core, so about 2M positions per window-day on two cores; expect two to three days.
3. Dedup the new chunks against data/all_unique.npz with train/dedup.py, then train two candidates with the same held-out set (data/held_d10.npz): net6's architecture (hidden 384, 20 epochs) and the 4/4 bucketed one from #44. Gate each with 200 fixed-node games, then 200 at 10+0.1; ship whichever is positive at 10+0.1.

Steps 1 and 2 are set up so that a day boundary only pauses them; step 3 uses train.py's per-epoch checkpoints.

**Comment by stefanobaghino** [comment]
Progress and a corrected estimate.

- Batch 1: 10,000 games generated this morning (5k nodes, `matches/bin/base44`), sampled to 1,098,924 positions (`data/selfplay4_1.fens`, every position after ply 8 except the last two). Relabelling at depth 10 started 13:22 in 22 chunks of 50k.
- Measured rate: chunk 0 took 33 min with two workers on cores 2-3, i.e. about 90k positions per hour, half the per-core figure quoted above (that one was for one worker per core with the other core free; two Stockfish workers on two cores share memory bandwidth and the bot's neighbours). A batch is therefore about 12 window-hours, one day each, four days for all four batches plus an hour of generation per batch.
- The pipeline (every step through `scripts/queue.sh`, ordered and window-gated) resumes by itself at 09:00 each day; nothing to babysit. A decision point comes after batch 2 (about 2.2M new positions, +50% over the current set): training the two candidates on that set first would give an answer two days earlier, with batches 3-4 continuing in the background. I will take that path unless the batch-2 relabel finishes faster than measured.

**Comment by stefanobaghino** [comment]
Batch-2 training is now automatic. `scripts/train_net8.sh` (794fb7e) waits until batches 1 and 2 are relabelled, then, behind the cores lock and inside the window: builds `data/selfplay4_unique.npz` with `train/dedup.py` (new depth-10 chunks first, then `data/all_unique.npz`, minus `data/held_d10.npz`), trains `net8a` (net6 architecture, main's trainer, so a win ships as a net-only release) and `net8k` (four king buckets, one output bucket, the `nnue-buckets` trainer and engine), 12 epochs each with the held-out loss picking the exported epoch, and gates each with 600 fixed-node games against a v0.1.46 build; candidates at or above zero get a 200-game 10+0.1 batch. Every step is skipped once its output exists, so a window stop just re-queues the job (a supervisor loop does that). Rehearsed today: the v0.1.46 baseline build and selfcheck, the dedup (about 500 bytes per position at peak, so 3 GB for 6.3M positions, within the machine's headroom), the trainer flags.

Expected timeline: batch 1 finishes its last 7 chunks this morning, batch 2 is generated and relabelled through tomorrow noon, and the candidates job takes about 7 window-hours, so the first numbers should land by tomorrow evening. Batches 3-4 continue behind it in the same queue. The main-trainer support for this (`dedup.py`, `--holdout`) came in through #56.

**Comment by stefanobaghino** [comment]
Batches 1 and 2 are relabelled and the candidates job (`scripts/train_net8.sh`) ran today on the deduplicated set: 5,856,673 unique positions (batches 1-2 at depth 10 plus the previous set, held-out positions excluded), 12 epochs, hidden 384.

| net | data / trainer | best holdout loss | vs v0.1.46, 600 games, 40k nodes |
|---|---|---|---|
| net8a | main's trainer, net6 architecture | 0.00749 | +106 ± 25 Elo |
| net8k | `nnue-buckets`, 4 king buckets, 1 output bucket | 0.00750 | +69 ± 25 Elo |

Timed gates at 10+0.1 (200 games, two concurrent on cores 2-3) are running; net8a stands at +147 ± 57 after 121 games. If it holds, net8a ships tonight as a net-only release: the engine code is unchanged since v0.1.46, so `engine/nets/default.bin` is the whole diff. Batch 3 is generated and its relabel is queued behind the gates; batch 4 follows.

**Comment by stefanobaghino** [comment]
Timed gates are final: net8a +145 ± 44 Elo vs v0.1.46 at 10+0.1 (200 games), net8k +67 ± 42. net8a shipped as v0.1.51 (net-only, engine code unchanged since v0.1.46).

Remaining for this ticket: batch 3 is generated and its relabel is running on cores 2-3 (23 chunks, it stops at 21:00 and resumes at 09:00), batch 4 follows; then a net9 on all four batches, same recipe, gated against v0.1.51.

**Comment by stefanobaghino** [comment]
Batch 3 is relabelled (22 chunks, finished 16:47 today at 26-28 min per chunk). Batch 4 was generated right after (10,000 games, 1,103,581 positions in 23 chunks) and its relabel started at 17:52; it stops at 21:00 and resumes at 09:00 tomorrow, so the four batches should be complete by Monday afternoon.

The net9 job (see #63) is queued behind it and starts as soon as the last chunk lands.

**Comment by stefanobaghino** [comment]
Stopped on request: the machine is being torn down.

State at the stop (2026-09-14 13:40):
- batches 1-3 fully relabelled at depth 10; batch 4 has 16 of 23 chunks (`data/selfplay4_d10_4_{0..15}.npz`), chunks 16-22 still missing.
- The relabel resumes from the first missing chunk with the same command behind the queue:
  `scripts/queue.sh --window --est 40 --name relabel4_4 -- scripts/relabel_chunks.sh data/selfplay4_4.fens data/selfplay4_d10_4 10 50000 2`
- Nothing of the net9 comparison has started (see #63).
